pinot-spi/src/main/java/org/apache/pinot/spi/metrics/PinotGauge.java

```diff
@@ -36,10 +38,26 @@
 
   Object getGauge();
 
+  /**
+   * Sets the gauge value.
+   * @param value the gauge value to set.
+   */
+  default void setValue(T value) {
+    throw new RuntimeException("setValue is not implemented");
+  }
+
+  /**
+   * Sets the gauge value supplier.
+   * @param valueSupplier the gauge value supplier to set.
+   */
+  default void setValueSupplier(Supplier<T> valueSupplier) {
```

Comment:
I wonder if it's possible to introduce a new interface that specifies the both setValue() and setValueSupplier() methods, and let this PinotGauge interface also extends the new interface you just created?

Reply:
that's a good suggestion! updated the code.